src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,965 @@
+            if (gridSize % 2 === 1) {
+                this.x = (gridSize / 2 - 1 / 2) * this.scalingFactor
+                this.y = (gridSize / 2 - 1 / 2) * this.scalingFactor
+            } else {
+                this.x = (gridSize / 2 - 1) * this.scalingFactor
+                this.y = (gridSize / 2) * this.scalingFactor
+            }
+
+            //The amount of numbers to the next turn increases every other turn.
+            this.numberToTurnAtForSpiral = 1
+            this.incrementForNumberToTurnAt = 1
+            this.whetherIncrementShouldIncrement = true
+        }
+    }
+
+    fillGridCell() {
+        this.drawBackgroundSquare()
+
+        this.drawPrimaryColorSquare()
+
+        this.drawSecondaryColorSquare()
+
+        this.showNumber()
+    }
+
+    drawBackgroundSquare() {
+        this.sketch.fill(this.backgroundColor)
+        this.drawBigSquare()
+    }
+
+    drawPrimaryColorSquare() {
+        this.colorMainColorProperties()
+        this.drawBigSquare()
+    }
+
+    drawSecondaryColorSquare() {
+        this.colorSecondaryColorProperties()
+        this.drawSmallSquare()
+    }
+
+    colorMainColorProperties() {
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization
+                    === PropertyVisualization.Primary_Color
+            ) {
+                if (
+                    this.hasProperty(
+                        this.currentNumber,
+                        this.propertyObjects[i].property
+                    )
+                ) {
+                    this.sketch.fill(this.propertyObjects[i].color)
+                }
+            }
+        }
+    }
+
+    colorSecondaryColorProperties() {
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization
+                    === PropertyVisualization.Secondary_Color
+            ) {
+                if (
+                    this.hasProperty(
+                        this.currentNumber,
+                        this.propertyObjects[i].property
+                    )
+                ) {
+                    this.sketch.fill(this.propertyObjects[i].color)
+                }
+            }
+        }
+    }
+
+    drawBigSquare() {
+        this.sketch.rect(
+            this.x,
+            this.y,
+            this.scalingFactor,
+            this.scalingFactor
+        )
+    }
+
+    drawSmallSquare() {
+        this.sketch.rect(
+            this.x + this.scalingFactor / 4,
+            this.y + this.scalingFactor / 4,
+            this.scalingFactor / 2,
+            this.scalingFactor / 2
+        )
+    }
+
+    hasProperty(num: bigint, property: Property) {
+        if (property === Property.Number_Sequence) {
+            for (
```

Comment:
Assuming this for loop is what you are using to compare two sequences in some of your presets, I think it may be somewhat inefficient. There's a javascript method includes() for an array. If you can pre-load the first 200 terms of this.seq into an array just once, then use the includes() method, I'm betting it would be faster? (I don't know for sure.)